src/components/filter/index.tsx

```diff
@@ -10,6 +10,7 @@ import { create as selectivelyCreate, Criteria } from "selectively"
 export class SmoothlyFilter {
 	freeSearchElement: HTMLSmoothlyInputElement
 	private inputs: Map<EventTarget, () => void> = new Map()
```

Comment:
since after Friday fika we learned that this might not be the best idea since re-renders might cause this components children (the event targets) to re-render causing this map to become larger and larger over time. i worked around this issue in the new form component so you can have a look there for a solution.